sympy/printing/repr.py

```diff
@@ -241,10 +241,18 @@ def _print_CoordinateSymbol(self, expr):
             )
 
     def _print_Predicate(self, expr):
-        return "%s(%s)" % (expr.__class__.__name__, self._print(expr.name))
+        name_repr = self._print(expr.name)
+        if expr.handlers:
+            handlers_repr = self._print(expr.handlers)
+            args_repr = ', '.join([name_repr, handlers_repr])
+        else:
+            args_repr = name_repr
+        return "%s(%s)" % (expr.__class__.__name__, args_repr)
 
     def _print_AppliedPredicate(self, expr):
-        return "%s(%s, %s)" % (expr.__class__.__name__, expr.func, expr.arg)
+        return "%s(%s)" % (
+            expr.__class__.__name__, self.reprify((expr.func,) + expr.args, ", ")
+        )
```

Comment:
Perhaps more readable as
```suggestion
        args = (expr.func,) + expr.args
        return "%s(%s)" % (expr.__class__.__name__, self.reprify(args, ", "))
```